eng/pipelines/aggregate-reports.yml

```diff
@@ -17,144 +22,166 @@ pr:
+            inputs:
+              pwsh: true
+              filePath: 'eng/scripts/dependencies/compare-deps-files.ps1'
+              arguments: >
+                -PSDepsFile 'https://aka.ms/ps-deps-stable'
+                -AzSdkDepsFile '$(Build.ArtifactStagingDirectory)/reports/azure-sdk.deps.json'
+          - task: PowerShell@2
+            displayName: 'Validate dependencies with pwsh preview'
+            inputs:
+              pwsh: true
+              filePath: 'eng/scripts/dependencies/compare-deps-files.ps1'
+              arguments: >
+                -PSDepsFile 'https://aka.ms/ps-deps-preview'
+                -AzSdkDepsFile '$(Build.ArtifactStagingDirectory)/reports/azure-sdk.deps.json'
+          - task: PublishPipelineArtifact@1
+            displayName: 'Publish Report Artifacts'
+            inputs:
+              artifactName: reports
+              path: '$(Build.ArtifactStagingDirectory)/reports'
+          - task: AzureFileCopy@2
+            displayName: 'Upload dependency report'
+            inputs:
+              sourcePath: '$(Build.ArtifactStagingDirectory)/reports'
+              azureSubscription: 'Azure SDK Artifacts'
+              destination: AzureBlob
+              storage: azuresdkartifacts
+              containerName: 'azure-sdk-for-net'
+              blobPrefix: dependencies
+          - task: PowerShell@2
+            displayName: "Verify Repository Resource Refs"
+            inputs:
+              pwsh: true
+              workingDirectory: $(Build.SourcesDirectory)
+              filePath: eng/common/scripts/Verify-Resource-Ref.ps1
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
+            displayName: 'Run CredScan'
+            inputs:
+              suppressionsFile: 'eng\CredScanSuppression.json'
+            condition: succeededOrFailed()
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+            displayName: 'Post Analysis'
+            inputs:
+              GdnBreakAllTools: false
+              GdnBreakGdnToolCredScan: true
+              GdnBreakGdnToolCredScanSeverity: Error
+              GdnBreakBaselineFiles: $(Build.SourcesDirectory)\eng\dotnet.gdnbaselines
+              GdnBreakBaselines: baseline  
+              # Used for generating baseline file
+              # GdnBreakOutputBaselineFile: dotnet
+              # GdnBreakOutputBaseline: baseline
+            condition: succeededOrFailed()
+            continueOnError: true
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
+            displayName: 'Publish Security Analysis Logs'
+            condition: succeededOrFailed()
+
+  - stage: PoliCheck
+    displayName: PoliCheck Evaluation
+    dependsOn: []
+    condition: ne(variables['ServiceDirectoryName'], '')
```

Comment:
Is this expected to be a queue time variable? Do we not run it over the entire repo all the time?

Reply:
Yes, If it is run over the entire repo then it will fail in every case, and the resulting failures will be very large So yes the PoliCheck will only run if a service directory is specified.

Reply:
By its nature aggregate-reports is supposed to run over the entire repo. We can make it not fail for until we get it cleaned up similar to what we are doing for CredScan.